src/Integration.UnitTests/MefServices/ActiveSolutionTrackerTests.cs

```diff
@@ -105,5 +107,31 @@ public void ActiveSolutionTracker_RaiseEventOnFolderOpen()
             counter.Should().Be(1, nameof(testSubject.ActiveSolutionChanged) + " was expected to be raised");
             isSolutionOpenArg.Should().BeTrue();
         }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void FolderWorkspaceInitializedEvent_RaiseEventOnProjectOpenedIfFolderWorkspace(bool isFolderWorkspace)
+        {
+            // Arrange
+            int counter = 0;
+            var testSubject = CreateTestSubject(isFolderWorkspace);
+            testSubject.FolderWorkspaceInitialized += (o, e) => { counter++; };
+
+            // Act
+            this.solutionMock.SimulateProjectOpen(null);
+
+            // Assert
+            counter.Should().Be(isFolderWorkspace ? 1 : 0);
+        }
+
+        private ActiveSolutionTracker CreateTestSubject(bool? isFolderWorkspace = null)
+        {
+            var folderWorkspaceFolder = new Mock<IFolderWorkspaceService>();
+            folderWorkspaceFolder.Setup(x => x.IsFolderWorkspace()).Returns(isFolderWorkspace ?? false);
```

Comment:
instead of a null-check it would make more sense to me to have false as the default value.